.gitignore

```diff
@@ -15,12 +15,44 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
-# OSX
+# macOS
 .DS_Store
 
 # Bazel
 bazel-*
 
-# IntelliJ
-.idea
-.ijwb
+# IntelliJ IDEA
+.idea/
+*.iml
+*.ipr
+*.iws
+*.geany
+.ijwb/
+
+# TypeDB Studio
+.typedb-studio
+
+# VS Code
+.vscode/
+.settings/
+.project
+.classpath
+
+# Eclipse
+.metadata/
+
+# Compiled
+dist/
+out/
+*.class
```

Comment:
I think none of these are relevant to us.
- `dist` is a common directory name where build artifacts are placed, but we don't use it in this repo;
- `out` is the default name IntelliJ IDEA uses for the compiler output directory, but we don't use it when building using Bazel
- `*.class` is relevant when compiling Java in the project directories themselves, but Bazel uses its sandbox

If these were included because they exist in `typedb` currently, I would suggest deleting them from `typedb`. I suspect they were in there because `typedb` used to be built using Gradle.

Reply:
These have been removed.